### PR TITLE
feat: create and list providers api

### PR DESCRIPTION
**Pull request by nnorbert**
### Ticket:

- [15123 ](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/15123)

### Description

Create backend API for creating and listing the providers.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [X] **New feature**
- [ ] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

N/A

### CODE REVIEW

File: apps/payments-backend/routes/providers/index.ts [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
     providerData: Type.Object({}),
+    status: Type.Union([
+      Type.Literal("connected"),
+      Type.Literal("disconnected"),
+    ]),
   }),
 ]);
 
-export type ProviderType = Static<typeof Provider>;
+export const ProvidersList = Type.Union([Type.Array(Provider)]);
```
Comment (alfonsograziano) [inline apps/payments-backend/routes/providers/index.ts:33]: Do we need to export all this stuff? Are we importing it somewhere?

File: apps/payments-backend/routes/providers/index.ts [hunk 2]
```diff
@@ -48,4 +67,58 @@ export default async function providers(app: FastifyInstance) {
...
+        },
+      },
+    },
+    async (request, reply) => {
+      const userId = request.user?.id;
+
+      const result = await app.pg.query(
+        `
+          SELECT
+            provider_id as "providerId",
```
Comment (alfonsograziano) [inline apps/payments-backend/routes/providers/index.ts:110]: Since the table is "payment_providers", shall we simplify the names using "id", "name", "type", "data"? We may want to do that in a different PR though, not related to this one

File: apps/payments/app/[locale]/(hosted)/paymentSetup/providers/add-banktransfer/page.tsx [hunk 3]
```diff
@@ -11,28 +11,16 @@ export default async () => {
...
-          accountNumber,
-          accountHolderName,
-        },
-      ],
-    );
+    await buildApiClient(userId).providers.apiV1ProvidersPost({
+      name: formData.get("provider_name") as string,
+      type: "banktransfer",
+      providerData: {
+        sortCode: formData.get("sort_code") as string,
```
Comment (alfonsograziano) [inline apps/payments/app/[locale]/(hosted)/paymentSetup/providers/add-banktransfer/page.tsx:19]: Since this is a generic object without further type enforcing, for now I guess we don't need to add `as string`

**Review by alfonsograziano — APPROVED** [review]
LGTM - apart from some optional comments

File: apps/payments-backend/routes/providers/index.ts (hunk 1 above)
Comment (nnorbert) [reply apps/payments-backend/routes/providers/index.ts:33]: Not at this moment. I will update it.

File: apps/payments-backend/routes/providers/index.ts (hunk 2 above)
Comment (nnorbert) [reply apps/payments-backend/routes/providers/index.ts:110]: Definitely, yes. Because the entity is provider, the properties inside should not repeat the entity's name. So I also think, we should use "id", "name", and so on.

File: apps/payments/app/[locale]/(hosted)/paymentSetup/providers/add-banktransfer/page.tsx (hunk 3 above)
Comment (nnorbert) [reply apps/payments/app/[locale]/(hosted)/paymentSetup/providers/add-banktransfer/page.tsx:19]: Sure, I will update it.

File: apps/payments-backend/routes/providers/index.ts (hunk 1 above)
Comment (nnorbert) [reply apps/payments-backend/routes/providers/index.ts]: Updated

File: apps/payments-backend/routes/providers/index.ts (hunk 2 above)
Comment (nnorbert) [reply apps/payments-backend/routes/providers/index.ts]: Updated

File: apps/payments/app/[locale]/(hosted)/paymentSetup/providers/add-banktransfer/page.tsx (hunk 3 above)
Comment (nnorbert) [reply apps/payments/app/[locale]/(hosted)/paymentSetup/providers/add-banktransfer/page.tsx]: Updated

**Review by alfonsograziano — APPROVED** [review]
